homeassistant/helpers/script.py

```diff
@@ -220,287 +236,349 @@ def referenced_entities(self):
         self._referenced_entities = referenced
         return referenced
 
-    def run(self, variables=None, context=None):
+    def run(self, variables=None, context=None, logger=None, message_base=None):
         """Run script."""
         asyncio.run_coroutine_threadsafe(
-            self.async_run(variables, context), self.hass.loop
+            self.async_run(variables, context, logger, message_base), self.hass.loop
         ).result()
 
-    async def async_run(
-        self, variables: Optional[Sequence] = None, context: Optional[Context] = None
-    ) -> None:
-        """Run script.
+    class _ScriptRun:
```

Comment:
Move class definition out of the other class definition.

Reply:
That's the way I had it at first, but it made reviewing the changes more difficult, so I moved it here. I can move it back out. Either way is fine by me.

Reply:
Let's move it out. It still makes changes difficult to review and it makes the code less easy to understand for future readers.